cmd/podman/artifact/rm.go

```diff
@@ -11,40 +11,62 @@ import (
 
 var (
 	rmCmd = &cobra.Command{
-		Use:               "rm ARTIFACT",
-		Short:             "Remove an OCI artifact",
-		Long:              "Remove an OCI from local storage",
-		RunE:              rm,
-		Aliases:           []string{"remove"},
-		Args:              cobra.ExactArgs(1),
+		Use:     "rm [options] ARTIFACT",
+		Short:   "Remove an OCI artifact",
+		Long:    "Remove an OCI from local storage",
+		RunE:    rm,
+		Aliases: []string{"remove"},
+		Args: func(cmd *cobra.Command, args []string) error { //nolint: gocritic
+			return checkAllAndArgs(cmd, args)
+		},
 		ValidArgsFunction: common.AutocompleteArtifacts,
-		Example:           `podman artifact rm quay.io/myimage/myartifact:latest`,
-		Annotations:       map[string]string{registry.EngineMode: registry.ABIMode},
+		Example: `podman artifact rm quay.io/myimage/myartifact:latest
+podman artifact rm -a`,
+		Annotations: map[string]string{registry.EngineMode: registry.ABIMode},
 	}
-	// The lint avoid here is because someday soon we will need flags for
-	// this command
-	rmFlag = rmFlagType{} //nolint:unused
+
+	rmOptions = entities.ArtifactRemoveOptions{}
 )
 
-// TODO at some point force will be a required option; but this cannot be
-// until we have artifacts being consumed by other parts of libpod like
-// volumes
-type rmFlagType struct { //nolint:unused
-	force bool
+func rmFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+	flags.BoolVarP(&rmOptions.All, "all", "a", false, "Remove all artifacts")
 }
-
 func init() {
 	registry.Commands = append(registry.Commands, registry.CliCommand{
 		Command: rmCmd,
 		Parent:  artifactCmd,
 	})
+	rmFlags(rmCmd)
 }
 
 func rm(cmd *cobra.Command, args []string) error {
-	artifactRemoveReport, err := registry.ImageEngine().ArtifactRm(registry.Context(), args[0], entities.ArtifactRemoveOptions{})
+	var nameOrID string
+	if len(args) > 0 {
+		nameOrID = args[0]
+	}
+	artifactRemoveReport, err := registry.ImageEngine().ArtifactRm(registry.Context(), nameOrID, rmOptions)
 	if err != nil {
 		return err
 	}
-	fmt.Println(artifactRemoveReport.ArtfactDigest.Encoded())
+	for _, d := range artifactRemoveReport.ArtfactDigests {
+		fmt.Printf("Deleted: %s\n", d.Encoded())
+	}
+	return nil
+}
+
+// checkAllAndArgs takes a cobra command and args and checks if
+// all is used, then no args can be passed. note: this was created
+// as an unexported local func for now and could be moved to pkg
+// validate.  if we add "--latest" to the command, then perhaps
+// one of the existing plg validate funcs would be appropriate.
+func checkAllAndArgs(c *cobra.Command, args []string) error {
+	all, _ := c.Flags().GetBool("all")
+	if all && len(args) > 0 {
+		return fmt.Errorf("--all cannot be combined with arguments")
```

Comment:
For consistency we should match the error messages from CheckAllLatestAndIDFile or at least the one from image rm, `when using the --all switch, you may not pass any images names or IDs` I think.